identity/src/main/java/com/stripe/android/identity/states/IDDetectorTransitioner.kt

```diff
@@ -99,33 +99,95 @@ internal class IDDetectorTransitioner(
         require(analyzerOutput is IDDetectorOutput) {
             "Unexpected output type: $analyzerOutput"
         }
-        return when {
-            timeoutAt.hasPassed() -> {
-                IdentityScanState.TimeOut(foundState.type, foundState.transitioner)
+        return when (analyzerOutput) {
+            is IDDetectorOutput.Legacy -> {
+                transitionFromFoundLegacy(
+                    foundState,
+                    analyzerOutput
+                )
             }
 
-            !outputMatchesTargetType(analyzerOutput.category, foundState.type) -> Unsatisfied(
-                "Type ${analyzerOutput.category} doesn't match ${foundState.type}",
+            is IDDetectorOutput.Modern -> {
+                transitionFromFoundModern(
+                    foundState,
+                    analyzerOutput
+                )
+            }
+        }
+    }
+
+    private fun transitionFromFoundLegacy(
```

Comment:
nit - there's some duplicated code across both transitions, maybe we can extract them and share somehow?

Reply:
hmm both transition logics are pretty unique in terms of the oder of checks, will keep it this way for better readability